test/modules.coffee

```diff
@@ -941,3 +941,57 @@ test "#4834: dynamic import", ->
       return bar = (await import('bar'));
     };
   """
+
+test "#5317: Support import.meta", ->
+  eqJS """
+    foo = import.meta
+  """,
+  """
+    var foo;
+
+    foo = import.meta;
+  """
+
+  eqJS """
+    foo = import
+        .meta
+  """,
+  """
+    var foo;
+
+    foo = import.meta;
+  """
+
+  eqJS """
+    foo = import.
+        meta
+  """,
+  """
+    var foo;
+
+    foo = import.meta;
+  """
+
+  eqJS """
+    foo = import.meta.bar
+  """,
+  """
+    var foo;
+
+    foo = import.meta.bar;
+  """
+
+  eqJS """
+    foo = import
+        .meta
+        .bar
+  """,
+  """
+    var foo;
+
+    foo = import.meta.bar;
+  """
+
+  throwsCompileError """
```

Comment:
This would be better expressed as a test in `test/error_messages.coffee` where you are specifying the exact error message thrown by the compiler

See the tests near the bottom of that file for `new.target` for examples - it looks to me like the `new.target outside of a function` test wouldn't apply to `import.meta`, but that the `new.target is only allowed meta property` (which is what you're testing here) and `new.target cannot be assigned` tests would apply to `import.meta`

Reply:
Done. Thanks!